### PR TITLE
WIP: Add azure-upi cucushift e2e tests

### PR DESCRIPTION
**Pull request by dis016**
[OCPQE-5687](https://issues.redhat.com/browse/OCPQE-5687) Add cucushift e2e test for azure-upi.
 re-opening [23220](https://github.com/openshift/release/pull/23220)
/cc @jhou1 @liangxia @pruan-rht @JianLi-RH

### CODE REVIEW

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__ocp-4.10.yaml [hunk 1]
```diff
@@ -152,6 +152,19 @@ tests:
...
+  interval: 24h
+  steps:
+    cluster_profile: azure4
+    env:
+      E2E_RUN_TAGS: '@azure-upi'
+      TAG_VERSION: '@4.10'
+    post:
+    - chain: upi-azurestack-post
+    pre:
+    - chain: upi-azurestack-pre
```
Comment (liangxia) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__ocp-4.10.yaml:165]: What's the diff between azurestack and azure ?

Comment (JianLi-RH) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__ocp-4.10.yaml:165]: @dis016 I found a page about cluster_profile, maybe you can try  ClusterProfile = "azurestack"?
https://github.com/openshift/ci-tools/blob/master/pkg/api/types.go#L1073

Comment (dis016) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__ocp-4.10.yaml:165]: upi on azure is based on azure stack https://azure.microsoft.com/en-us/products/azure-stack/hub/

File: ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__ocp-4.10.yaml [hunk 2]
```diff
@@ -152,6 +152,19 @@ tests:
     - chain: upi-vsphere-pre
     - ref: cucushift-pre
     workflow: cucushift-e2e
+- as: e2e-azure-cucushift-upi
```
Comment (liangxia) [inline ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__ocp-4.10.yaml:155]: Perhaps we need `azurestack` there to identy that ?

Comment (dis016) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__ocp-4.10.yaml:155]: @liangxia yes, if the tests succeed, i will fix the naming conventions to azurestack

Comment (jhou1) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__ocp-4.10.yaml:155]: Is azurestack the `Azure Stack Hub`?

Comment (dis016) [reply ci-operator/config/openshift/verification-tests/openshift-verification-tests-master__ocp-4.10.yaml:155]: @jhou1 yes, i guess azure upi is relying on Azure Stack Hub.

**Comment by dis016** [comment]
New PR will get opened

**Pull request closed without merging**